library/bignum.c

```diff
@@ -167,6 +178,9 @@ int mbedtls_mpi_shrink( mbedtls_mpi *X, size_t nblimbs )
     if( i < nblimbs )
         i = nblimbs;
 
+    if( X->fixedbuf )
```

Comment:
Surely this is not needed? IIUC this function is not obligated to reduce the size ("Resize down as much as possible", line 156), and by this point `X` doesn't need to grow, so it can just be left as-is.

Reply:
Good point -- it depends on what you want `fixedbuf` to achieve: It's meant to flag and fail upon _any_ attempt to resize, not just when there's no other way. If you mark an MPI as having a fixed size buffer and still call `mbedtls_mpi_shrink()` on it, that's a bug. Note that in contrast to `mbedtls_mpi_grow()`, `mbedtls_mpi_shrink()` is not implicitly called by other MPI API (luckily!).